File: apps/web/shared/types/audit.ts
/** Audit run status / poll payload (API snake_case). */

export type AuditStatus =
  | "pending"
  | "running"
  | "complete"
  | "complete_with_warnings"
  | "failed"
  | "cancelled"
  | string;

export type AuditCreateRequest = {
  website_id: string;
};

export type AuditCreateResponse = {
  audit_id: string;
  status: AuditStatus;
  message: string;
};

export type EngineSummaryItem = {
  engine?: string;
  engine_name?: string;
  status: string;
  duration_ms?: number | null;
  started_at?: string | null;
  completed_at?: string | null;
  error_message?: string | null;
};

export type AuditPollResponse = {
  audit_id: string;
  website_id: string;
  url: string;
  canonical_url?: string | null;
  status: AuditStatus;
  progress: number;
  current_engine: string | null;
  health_score?: number | null;
  scores?: Record<string, number> | null;
  category_scores?: Record<string, number> | null;
  engine_summary?: EngineSummaryItem[];
  failure_code?: string | null;
  failure_message?: string | null;
  duration_ms?: number | null;
  started_at?: string | null;
  completed_at?: string | null;
  created_at?: string;
  updated_at?: string;
};

export function isTerminalAuditStatus(status: string): boolean {
  return (
    status === "complete" ||
    status === "complete_with_warnings" ||
    status === "failed" ||
    status === "cancelled"
  );
}

export function isSuccessfulAuditStatus(status: string): boolean {
  return status === "complete" || status === "complete_with_warnings";
}

File: apps/api/app/schemas/audit.py
"""Audit Run API schemas — API_SPEC §6.1 / §6.3 (Sprint 3 subset)."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class AuditCreateRequest(BaseModel):
    website_id: UUID


class AuditCreateResponse(BaseModel):
    audit_id: UUID
    status: str
    message: str = "Audit created successfully."


class AuditScoresResponse(BaseModel):
    overall: int | None = None
    seo: int | None = None
    performance: int | None = None
    security: int | None = None
    accessibility: int | None = None
    business: int | None = None
    roi: int | None = None


class AuditDetailResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    audit_id: UUID
    website_id: UUID
    url: str
    canonical_url: str
    status: str
    progress: int = Field(description="0–100 progress percent")
    current_engine: str | None = Field(
        default=None,
        description="Current engine name; maps to API_SPEC current_step",
    )
    scores: AuditScoresResponse
    failure_code: str | None = None
    failure_message: str | None = None
    started_at: datetime | None = None
    completed_at: datetime | None = None
    created_at: datetime
    updated_at: datetime
